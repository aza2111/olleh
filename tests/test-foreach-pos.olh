//test-foreach-pos: testing foreach loops


foreach x ['a','b','c']:
        print x;
end;

foreach ey [['a','b','c'],['d']]:
        print ey;
end;

map<char> m1 = {'x'->0};
m1.set('y', 4);
print m1.getLength();
foreach yy m1:
        print yy;
        print m1.get(yy);
end;


map<String> m = fresh map(String);

m.set("example", 5);
print m.getLength();
print m.getLength();
m = { "a" -> 3, "egxample" -> 5};
print m.getLength();
m.set("anotherr example", -9);
print m.getLength();
foreach z m:
        //print m.get(z);
        print z + " " +  m.get(z);
end;
