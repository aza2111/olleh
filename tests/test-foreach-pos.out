'a'
'b'
'c'
['a', 'b', 'c']
['d']
2
'x'
0
'y'
4
1
1
2
3
egxample 5
a 3
anotherr example -9
